src/loadMessages.js

```diff
@@ -0,0 +1,11 @@
+export default function loadMessages(lang) {
+  const wrapped = import(`./messages/${lang}`)
+
+  if (typeof window !== 'undefined') {
+    return wrapped.then(msgs => msgs.default)
+  }
+
+  /* eslint-disable global-require, import/no-dynamic-require, prefer-template */
+  // return Promise.resolve(require('./messages/' + lang).default)
+  return __webpack_require__(wrapped.resolve()).default
+}
```

Comment:
the problem is this: `wrapped.resolve` won't work with the babel plugin. I didnt even know promises had that method. I dont think they do. i just did it in the chrome canary console. `Promise.resolve` exists, but not `(new Promise((resolve, reject) => resolve())).resolve`

the babel plugin fakes the `.then` and `.catch` methods. So if you did the following it would work and is likely the answer you're looking for:

`return wrapped.then(msgs => msgs.default)`

that's if it works server-side like ur saying. but if u need a quick fix, just do resolveWeak and wrap it in `Promise.resolve` until i get a chance to take a look at this.